cat-launcher/src-tauri/src/settings/list_themes.rs

```diff
@@ -0,0 +1,118 @@
+use std::collections::HashSet;
+use std::io;
+use std::path::{Path, PathBuf};
+
+use serde_json;
+use strum::{IntoEnumIterator, IntoStaticStr};
+
+use crate::active_release::repository::{
+  ActiveReleaseRepository, ActiveReleaseRepositoryError,
+};
+use crate::filesystem::paths::{
+  get_game_resources_dir, GetGameExecutableDirError,
+};
+use crate::infra::utils::{OSNotSupportedError, OS};
+use crate::settings::{ColorDefWrapper, ColorTheme};
+use crate::variants::GameVariant;
+
+#[derive(thiserror::Error, Debug, IntoStaticStr)]
+pub enum ListThemesError {
+  #[error("failed to get OS: {0}")]
+  OSNotSupported(#[from] OSNotSupportedError),
+
+  #[error("failed to read directory: {0}")]
+  ReadDirectory(#[from] io::Error),
+
+  #[error("failed to get game resources dir: {0}")]
+  GameResourcesDirectory(#[from] GetGameExecutableDirError),
+
+  #[error("failed to parse theme file: {0}")]
+  ParseFile(#[from] serde_json::Error),
+
+  #[error("failed to get active release: {0}")]
+  ActiveRelease(#[from] ActiveReleaseRepositoryError),
+}
+
+pub async fn list_themes(
+  data_dir: PathBuf,
+  repository: &impl ActiveReleaseRepository,
+  os: &OS,
+) -> Result<Vec<ColorTheme>, ListThemesError> {
+  let mut themes = Vec::new();
+
+  for variant in GameVariant::iter() {
+    let variant_themes =
+      get_themes_for_variant(&variant, &data_dir, repository, os)
+        .await;
+    themes.extend(variant_themes);
+  }
+
+  // Remove duplicates
+  let mut seen = HashSet::new();
+  themes.retain(|theme| seen.insert(theme.name.clone()));
+
+  Ok(themes)
+}
+
+async fn get_themes_for_variant(
+  variant: &GameVariant,
+  data_dir: &Path,
+  repository: &impl ActiveReleaseRepository,
+  os: &OS,
+) -> Vec<ColorTheme> {
+  let release = match repository.get_active_release(variant).await {
+    Ok(Some(release)) => release,
+    _ => return Vec::new(),
+  };
+
+  let resources_dir =
+    match get_game_resources_dir(variant, &release, data_dir, os)
+      .await
+    {
+      Ok(dir) => dir,
+      _ => return Vec::new(),
+    };
+
+  let themes_dir =
+    resources_dir.join("data").join("raw").join("color_themes");
+
+  if !themes_dir.exists() {
+    return Vec::new();
+  }
+
+  let mut entries = match tokio::fs::read_dir(themes_dir).await {
+    Ok(entries) => entries,
+    _ => return Vec::new(),
+  };
+
+  let mut themes = Vec::new();
+  while let Ok(Some(entry)) = entries.next_entry().await {
+    if let Some(theme) = parse_theme_file(&entry.path()).await {
+      themes.push(theme);
+    }
+  }
+
+  themes
+}
+
+async fn parse_theme_file(path: &Path) -> Option<ColorTheme> {
+  if !path.is_file() {
```

Comment:
In cat-launcher/src-tauri/src/settings/list_themes.rs around lines 102 to 104,
the code uses the synchronous path.is_file() inside an async context; replace
that call with an async filesystem metadata lookup (e.g.
tokio::fs::metadata(path).await) and check the returned Metadata.is_file(),
handling errors by returning None (or mapping errors to None) so the function
remains async-safe and does not block the runtime.